build-ci/github_protect_travis_flake8.rst

```diff
@@ -0,0 +1,79 @@
+#############################################
+Github Branch Protection, TravisCI and Flake8
+#############################################
+
+DM recommends using `Github branch protection <https://help.github.com/articles/about-protected-branches/>`_, `TravisCI <https://travis-ci.org/>`_ and `Flake8 <http://flake8.pycqa.org/en/stable/>`_ on repositories.
+
+.. _branch-protection:
+
+Github Branch Protection
+========================
+
+All DM repositories should protect the master branch. Protected branches block some features of Git. Specifically DM protected branches can't be edited from the web, force pushed, deleted or merged until their status checks pass. Furthermore status checks must pass using the latest base branch code.
+
+.. _travisci:
+
+Travis CI
+=========
+
+TravisCI is a service that provides continuous integration for Github repositories. This service must be enabled for the repository before it will work. After it is enabled there will be a page available with the TravisCI status for the repository. For example lsst.utils `https://travis-ci.org/lsst/utils <https://travis-ci.org/lsst/utils>`_.
+
+.. _flake8:
+
+Flake8
```

Comment:
I believe @jdswinbank would like our Travis documentation to include descriptions of our Travis jobs. Out of scope for this ticket but we should design the layout of this document with that in mind (so the lsstsw checking, the latex/sphinx checking, flake8 checking).